sklearn/metrics/_plot/det_curve.py

```diff
@@ -300,6 +301,13 @@ def plot(self, ax=None, *, name=None, **kwargs):
         line_kwargs = {} if name is None else {"label": name}
         line_kwargs.update(**kwargs)
 
+        # avoid inf from sp.stats.norm.ppf
+        eps = 1e-8
+        self.fpr = np.where(self.fpr == 0, self.fpr + eps, self.fpr)
+        self.fpr = np.where(self.fpr == 1, self.fpr - eps, self.fpr)
+        self.fnr = np.where(self.fnr == 0, self.fnr + eps, self.fnr)
+        self.fnr = np.where(self.fnr == 1, self.fnr - eps, self.fnr)
```

Comment:
This could probably be simplified with things like `self.fpr = self.fpr.clip(eps, 1 - eps)` or similar.